tests/lib/util/node-event-generator.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Helpers to check EventGenerator interface.
+ * @author Toru Nagashima
+ * @copyright 2015 Toru Nagashima. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var assert = require("assert");
+var NodeEventGenerator = require("../../../lib/util/node-event-generator");
+var helper = require("./event-generator-helper");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+describe("NodeEventGenerator", function() {
+    helper.testEventGeneratorInterface(
+        new NodeEventGenerator(helper.createDummyEmitter())
```

Comment:
Instead of doing a dummy emitter, create a real one and use sinon.mock() to make sure emit() is called with the correct arguments. That way, you can have separate tests for enterNode and leaveNode , too.